src/shared/services/UserService.ts

```diff
@@ -48,7 +48,15 @@ export class UserService {
     this.myUserInfo = undefined;
     IsomorphicCookie.remove("jwt"); // TODO is sometimes unreliable for some reason
     document.cookie = "jwt=; Max-Age=0; path=/; domain=" + location.hostname;
-    location.reload();
+    if (
+      /create_.*|inbox|settings|setup|admin|reports|registration_applications/g.test(
```

Comment:
It might be worth extracting this into a function and putting it in utils. It would help with what I called out [on another PR](https://github.com/LemmyNet/lemmy-ui/pull/1013#discussion_r1192991566).